vhost/src/vhost_user/dummy_backend.rs

```diff
@@ -262,6 +263,17 @@ impl VhostUserBackendReqHandlerMut for DummyBackendReqHandler {
     #[cfg(feature = "gpu-socket")]
     fn set_gpu_socket(&mut self, _gpu_backend: GpuBackend) {}
 
+    fn get_shared_object(&mut self, _uuid: VhostUserSharedMsg) -> Result<OwnedFd> {
+        let file = tempfile::tempfile().unwrap();
+
+        // SAFETY: We are calling `OwnedFd::from_raw_fd` to take ownership of the raw file descriptor.
+        // This is safe in this context because:
+        // 1. The file descriptor returned by `into_raw_fd()` is valid and open.
```

Comment:
But are we sure it remains open after the end of this block?

For example in `get_inflight_fd()` we stored it in `self.inflight_file`, so IMO we should do something similar, otherwise the file will be closed and the `fd` will be invalid just after the call of `get_shared_object`.

Reply:
right, I missed the `into_raw_fd()` call. In that case, it consumes it, so the OwnedFd will take the ownership!